ntp-daemon/src/main.rs

```diff
@@ -23,5 +27,66 @@ async fn main() -> Result<(), Box<dyn Error>> {
     // this away when not using sentry.
     let _guard = finish_tracing_init(&mut config, has_log_override)?;
 
-    ntp_daemon::spawn(&config.system, &config.peers).await
+    let peers_reader = Arc::new(tokio::sync::RwLock::new(Vec::new()));
+    let peers_writer = peers_reader.clone();
+
+    std::fs::remove_dir_all("/run/ntpd-rs")?;
+    std::fs::create_dir("/run/ntpd-rs")?;
+
+    let peers_listener = UnixListener::bind("/run/ntpd-rs/observe")?;
+
+    // this binary needs to run as root to be able to adjust the system clock
+    // directories inherit root permissions, but the client should not need elevated
+    // permissions to read from the socket. So we explicitly set the permissions
+    let permissions: std::fs::Permissions = PermissionsExt::from_mode(0o777);
```

Comment:
so maybe this makes perfect sense, but I needed `7` (so read/write/execute) as the permission here. intuitively 6 (read/write) should be enough but that did not work.

Reply:
not sure the permissions make sense to me either. However, I would suggest we also make this configurable so that a system administrator can decide who should be able to access the socket.